src/fusion-order/fusion-order.ts

```diff
@@ -331,14 +315,14 @@ export class FusionOrder {
                 // shift because of how LimitOrder.buildSalt works
                 salt: BigInt(order.salt) >> 160n,
                 maker: new Address(order.maker),
-                receiver: new Address(order.receiver),
+                receiver: extra?.customReceiver,
```

Comment:
Here we need to provide either real receiver (or 0x0000) or settlementExtension in case fees > 0